hashicorp/consul-k8s
# Default namespaces for CRDs that support namespace when enabled

Changes proposed in this PR:
- Default the namespace fields of resources when namespaces are enabled so that the controller doesnt continuously attempt to reconcile them as the value presented by consul does not match the spec (because Consul defaults the namespace field where we don't).

How to test this PR:
- Code Review
- Deploy `consul-enterprise` with the consul-k8s image `ashwinvenkatesh/consul-k8s:default-ns`. Allow namespaces on the `consul` installation.

Verify that creating an `ingress-gateway`, `terminating-gateway` and `service-router` without namespace fields reconciles just once and does not lead to a reconcile loop.

Checklist:
